### PR TITLE
Generate JumpProblem with EvalFunc to avoid GG

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Does the same change as https://github.com/SciML/ModelingToolkit.jl/pull/451 except on JumpSystem to fix https://github.com/SciML/ModelingToolkit.jl/issues/452

### CODE REVIEW

File: src/systems/jumps/jumpsystem.jl [hunk 1]
```diff
@@ -139,11 +143,13 @@ tspan = (0.0, 250.0)
 dprob = DiscreteProblem(js, u₀map, tspan, parammap)
 ```
 """
-function DiffEqBase.DiscreteProblem(sys::AbstractSystem, u0map, tspan::Tuple,
+function DiffEqBase.DiscreteProblem(sys::JumpSystem, u0map, tspan::Tuple,
```
Comment (ChrisRackauckas) [inline src/systems/jumps/jumpsystem.jl:146]: @isaacsas generally we should leave DiscreteProblem open for discrete dynamical systems. I want to add a differencing operator and handle building these, but with JumpProblem we know this is always identity so we can have this special dispatch.

File: src/systems/jumps/jumpsystem.jl [hunk 2]
```diff
@@ -139,11 +143,13 @@ tspan = (0.0, 250.0)
...
                                     parammap=DiffEqBase.NullParameters(); kwargs...)
     u0 = varmap_to_vars(u0map, states(sys))
     p  = varmap_to_vars(parammap, parameters(sys))
-    f  = (du,u,p,t) -> du.=u    # identity function to make syms works
+    # identity function to make syms works
+    # EvalFunc because we know that the jump functions are generated via eval
+    f  = DiffEqBase.EvalFunc(DiffEqBase.DISCRETE_INPLACE_DEFAULT)
```
Comment (ChrisRackauckas) [inline src/systems/jumps/jumpsystem.jl:152]: This is just a special trick so if you use `FunctionMap` it knows to ignore handling the discrete function map part.

**Comment by ChrisRackauckas** [comment]
@TorkelE was this the last issue for getting DiffEqBio working?

**Comment by TorkelE** [comment]
Yes. This should make all my tests succeed.

There are some tangential things which are not implemented yet:
- Network property query. We already get everything that works on `ReactionSystem`s in MTK, so not sure what more we would want to add here.
- Not sure what we decided on a `merge!(rs1,rs2)` function was, but if that gets implemented in MTK, a way for calling that from DiffEqBio needs implementing (likely trivial).
- Latexify support.
- Scaling the noise in the chemical Langevin equation. This might actually need some decision on how to parse additional input, so not yet supported.

But if we want to, we could probably merge without these points fixed.

**Comment by ChrisRackauckas** [comment]
>Network property query. We already get everything that works on ReactionSystems in MTK, so not sure what more we would want to add here.

I think we're good here then.

>Not sure what we decided on a merge!(rs1,rs2) function was, but if that gets implemented in MTK, a way for calling that from DiffEqBio needs implementing (likely trivial).

That's through the same system as components. I think I need to talk to @isaacsas about it, but don't worry about it in DEBio quite yet.

>Latexify support.

@korsbo we should add a special Latexify recipe for ReactionSystem. Latexify recipes already exist on other systems, but the arrow notation is nice. This is https://github.com/SciML/ModelingToolkit.jl/issues/288 , so we'd just need to update https://github.com/SciML/DiffEqBiological.jl/blob/master/src/latexify_recipes.jl

>Scaling the noise in the chemical Langevin equation. This might actually need some decision on how to parse additional input, so not yet supported.

I am not sure if we should generally do this. We should discuss other transformations, like LNA and such, but these are transforms on the ReactionSystem.

**Comment by ChrisRackauckas** [comment]
So I think we're good then. @isaacsas and I should have a chat about composing though.

**Comment by TorkelE** [comment]
Sounds good. I agree there is some dubiousness with scaling noise in CLA, which would require further discussion.

Regarding system components, when I had a look at it earlier I allowed the macro without input (`rn = @reaction_network` to create an empty `ReactionSystem`, which then could be built upon. This worked fine, but it seemed that someway along with the development of MTK that was disallowed, and creating empty `ReactionSystem`s now yields an error. Depending on how this goes this might be something to consider.

**Comment by isaacsas** [comment]
I’d like to get the relevant subset of the API updated before we release so if @TorkelE is done I can take over the PR.

**Comment by isaacsas** [comment]
Alternatively we could merge it to master and then I can update and release.

**Comment by ChrisRackauckas** [comment]
Which PR?

**Comment by isaacsas** [comment]
His DEBio one. Not here.

**Comment by ChrisRackauckas** [comment]
I think it could be good to just tidy it up and merge to master, but not release it yet. Then do some updates via other PRs before releasing. PRs to PRs can go too deep and just cause more of a mess IMO.

**Comment by TorkelE** [comment]
Yes, let's merge and then @isaacsas can fix the last few parts. There has to be a new release of this for Travis to load the latest version, right? When the tests have let's merge.

**Comment by ChrisRackauckas** [comment]
It's tagged.

**Pull request merged**